From d4fc56c8dbf9f75a21345075305a17438443e1ed Mon Sep 17 00:00:00 2001
From: Sankalp Negi <snegi@qti.qualcomm.com>
Date: Thu, 22 Jan 2026 21:50:32 -0800
Subject: [PATCH 1/2] PENDING: kallsyms: Export kallsyms_lookup_name

Export kallsyms_lookup_name() to allow legitimate in-kernel users
(such as minidump and debug infrastructure) to resolve core kernel
symbols that are not explicitly exported.

This avoids the need to export multiple individual symbols and
provides a unified, controlled mechanism to look up internal kernel
symbols when required.

Change-Id: I9d5b6f3c6c07ee5e800de3e8ab1c5877953631db
Signed-off-by: Sankalp Negi <snegi@qti.qualcomm.com>
---
 kernel/kallsyms.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/kernel/kallsyms.c b/kernel/kallsyms.c
index 18edd57b5fe8..3283f090d421 100644
--- a/kernel/kallsyms.c
+++ b/kernel/kallsyms.c
@@ -277,6 +277,7 @@ unsigned long kallsyms_lookup_name(const char *name)
 
 	return module_kallsyms_lookup_name(name);
 }
+EXPORT_SYMBOL_GPL(kallsyms_lookup_name);
 
 /*
  * Iterate over all symbols in vmlinux.  For symbols from modules use
-- 
2.34.1

